### DISCUSSION TITLE
Layer swipe with multiple layers

### DISCUSSION THREAD

**Discussion by gtnbssn** [openlayers/openlayers#15217] [Q&A]
I am implementing the layer swipe following the official example: https://openlayers.org/en/latest/examples/webgl-layer-swipe.html

I have a number of COG layers over the OSM basemap, and the interface allows users to rearrange the z-index of each layer to decide what goes on top.

My goal is to be able to let the user decide where in the stack of layers the swipe will cut, so we could have 5 layers that are cut and the rest under fully visible for instance.

I am running into some issues though, maybe similar to this open bug: https://github.com/openlayers/openlayers/issues/14641

Has anyone managed to do this before? I am particularly confused about how the official example is working without calling `gl.clear()`, which I have to call in my current experiment, but seems to be the reason why I cannot have multiple layers.

**Comment by jahow** [comment]
As answered in the issue (which I saw before this discussion), this is a problem that will be recurrent when trying to do a swipe/split effect with multiple WebGL layers.

I have investigated this issue quite a bit already and hopefully will be able to come up with a fix soon.
